src/extension.ts

```diff
@@ -1,173 +1,224 @@
+        vscode.workspace.createFileSystemWatcher("**/*.txt"),
+        vscode.workspace.createFileSystemWatcher("**/*.yaml"),
+        vscode.workspace.createFileSystemWatcher("snooty.toml"),
+      ],
+    },
+  };
+
+  const client = new LanguageClient(
+    "Snooty Language Client",
+    serverOptions,
+    clientOptions
+  );
+  const restartServer = vscode.commands.registerCommand(
+    "snooty.restart",
+    async () => {
+      await client.stop();
+      return client.start();
     }
-
-    const run: Executable = {
-        args: ['language-server'],
-        command: executableCommand
-    };
-    const debug: Executable = run;
-    const serverOptions: ServerOptions = {
-        run: run,
-        debug: debug
-    };
-
-    // client extensions configure their server
-    const documentSelector = [
-        { language: 'plaintext', scheme: 'file' },
-        { language: 'yaml', scheme: 'file' },
-        { language: 'restructuredtext', scheme: 'file' },
-        { language: 'toml', scheme: 'file' },
-    ];
-    const clientOptions: LanguageClientOptions = {
-        documentSelector,
-        synchronize: {
-            configurationSection: 'snooty',
-            fileEvents: [
-                vscode.workspace.createFileSystemWatcher('**/*.rst'),
-                vscode.workspace.createFileSystemWatcher('**/*.txt'),
-                vscode.workspace.createFileSystemWatcher('**/*.yaml'),
-                vscode.workspace.createFileSystemWatcher('snooty.toml')
-            ]
-        }
+  );
+
+  // Push the disposable to the context's subscriptions so that the
+  // client can be deactivated on extension deactivation
+  context.subscriptions.push(client.start());
+  context.subscriptions.push(restartServer);
+
+  // Register custom command to allow includes, literalincludes, and figures to be clickable
+  const clickInclude: vscode.Disposable = vscode.commands.registerCommand(
+    "snooty.clickInclude",
+    async (args) => {
+      const hoverFile = args.hoverFile;
+
+      // Send request to server (snooty-parser)
+      const type = mime.getType(hoverFile);
+
+      if (type == null || !type.includes("image")) {
+        const textDoc = await vscode.workspace.openTextDocument(hoverFile);
+        vscode.window.showTextDocument(textDoc);
+      } else {
+        open(hoverFile);
+      }
     }
+  );
+  context.subscriptions.push(clickInclude);
+
+  // Shows clickable link to file after hovering over it
+  vscode.languages.registerHoverProvider(
+    documentSelector,
+    new (class implements vscode.HoverProvider {
+      provideHover(
+        _document: vscode.TextDocument,
+        _position: vscode.Position,
+        _token: vscode.CancellationToken
+      ): vscode.ProviderResult<vscode.Hover> {
+        // Get range for a link
+        const wordRegex = /\/\S+/;
+        const wordRange = _document.getWordRangeAtPosition(
+          _position,
+          wordRegex
+        );
```

Comment:
(Non-blocking comment) Based on the generic regex used for hovering, doc role links are captured and the resulting hover content ends up looking a little weird (see image). This is most likely out of scope since I think this is partially existing behavior (with the long query param being a new addition). 

Just wanted to call this out in case this is something that seems like we should file a separate ticket for to fix.

<img width="1148" alt="Screen Shot 2024-05-13 at 6 38 50 PM" src="https://github.com/mongodb/snooty-vscode/assets/27821750/e08c28b7-1c55-4d52-8e5f-6c59b22d591c">

Reply:
thanks for catching this! will file a ticket once this one is merged